pkg/operator/rules_test.go

```diff
@@ -243,12 +243,25 @@ func shouldDropKeepFiringForThanos(t *testing.T) {
 			},
 		}},
 	}
-
-	thanosVersion, _ := semver.ParseTolerant(DefaultThanosVersion)
-	pr := newRuleSelectorForConfigGeneration(ThanosFormat, thanosVersion)
-	content, _ := pr.generateRulesConfiguration(rules)
-	if strings.Contains(content, "keep_firing_for") {
-		t.Fatalf("expected `keep_firing_for` not to be present in PrometheusRule")
+	testCases := []struct {
```

Comment:
The test doesn't work as expected: the second sub-test should return a snippet with `keep_firing_for`. It's because generateRulesConfiguration() mutates the rules struct.
Also the `shouldDropKeepFiringForThanos` test function needs to be renamed or a new test function should be added.

Reply:
I hope I got your point. 
Changes were pushed.